crates/codegen/parser/runtime/src/support/sequence_helper.rs

```diff
@@ -107,6 +109,52 @@ impl SequenceHelper {
                         next.expected_tokens,
                     ));
                 }
+                // Enter recovery mode
+                (ParserResult::Match(running), ParserResult::SkippedUntil(mut skipped)) => {
+                    running.nodes.extend(std::mem::take(&mut skipped.nodes));
+                    self.result = State::Running(ParserResult::SkippedUntil(SkippedUntil {
+                        nodes: std::mem::take(&mut running.nodes),
+                        ..skipped
+                    }));
+                }
+                (ParserResult::PrattOperatorMatch(_), ParserResult::SkippedUntil(_)) => todo!(),
+                // See if we can recover
+                (ParserResult::SkippedUntil(running), ParserResult::Match(next)) => {
+                    let tokens: Vec<_> =
+                        next.nodes.iter().filter_map(cst::Node::as_token).collect();
+                    let mut rules = next.nodes.iter().filter_map(cst::Node::as_rule);
+
+                    let is_single_token_with_trivia =
+                        tokens.len() == 1 && rules.all(|rule| rule.kind.is_trivia());
+                    let found_token = tokens.first().copied();
+
+                    // We can only recover if the next node is a single token (with trivia) that's expected by us
+                    // or an ancestor
+                    if is_single_token_with_trivia
+                        && found_token.map_or(false, |t| running.expected_tokens.contains(&t.kind))
+                    {
```

Comment:
This is a bit hacky as this couples a bit the recovery and the sequence helper logic here; Also trivia handling seems wrong when I tried to plug this same mechanism to the TerminatedBy. Will need to investigate further, but maybe it's good enough for just DelimitedBy for now?